MSAL/src/requests/MSALSilentRequest.m

```diff
@@ -142,25 +144,150 @@ - (void)acquireTokenImpl:(MSALCompletionBlock)completionBlock
     }
 
     NSError *msidError = nil;
+    MSIDAppMetadataCacheItem *appMetadata = [self.tokenCache getAppAppMetadataForConfiguration:msidConfiguration
+                                                                                       context:_parameters
+                                                                                         error:&msidError];
+    if (msidError)
+    {
+        completionBlock(nil, msidError);
+        return;
+    }
+    //On first network try, app metadata will be nil but on every subsequent attempt, it should reflect if clientId is part of family
+    NSString *familyId = appMetadata ? appMetadata.familyId : @"1";
+    if (![NSString msidIsStringNilOrBlank:familyId])
+    {
+        [self tryFRT:familyId appMetadata:appMetadata completionBlock:completionBlock];
+    }
+    else
+    {
+        [self tryMRRT:completionBlock];
+    }
+}
 
-    self.refreshToken = [self.tokenCache getRefreshTokenWithAccount:_parameters.account.lookupAccountIdentifier
-                                                           familyId:nil
-                                                      configuration:msidConfiguration
-                                                            context:_parameters
-                                                              error:&msidError];
+- (MSIDRefreshToken *)getRefreshToken:(NSString *)familyId error:(NSError **)error
+{
+    _familyId = familyId;
```

Comment:
I didn't mean to change it from property to instance variable. I was wondering why do we need to store it since we're passing it to our methods?